### ISSUE TITLE
Meson tries to find VAPI for Vala project's C dependencies' dependencies

### ISSUE THREAD

**Issue by liferooter** [mesonbuild/meson#9959]
**Describe the bug**
I have Vala project. It depends on C library. This library also depends on other C libraries which are presented on my computer, but building fails saying that no VAPI found for them. There is no reason to require VAPI for C dependency's C dependencies.

**To Reproduce**
My project: https://github.com/liferooter/textpieces/tree/4eefcda6d4b78741fd88b1ade9c2ee3bb4274ec3
It requires `gtksourceview-5` which requires some C-only libraries (`libcre2-8` and other), but can't be built because there are no VAPI for these libraries.

**Expected behavior**
Meson should successfully build Vala projects with C dependencies with their own C-only dependencies.

**System parameters**
* **OS:** Fedora Linux 35
* **Python version:** 3.10.2
* **Meson version:** 0.59.4
* **Ninja version:** 1.10.2

**Comment by nda-cunh** [comment]
it would be nice to have an argument like novapi for
```meson
dependency('my_lib', novapi: true)
```

**Comment by dcbaker** [comment]
We shouldn't be searching for vapi by default, we have a `language : ...` parameter for `dependency()`, it just is only used in a handful of places, and we should change `dependency('my_lib', language : 'vala')` to get vala interfaces, and `dependency('my_lib')` to just get the C/C++/whatever dependency.

**Comment by dcbaker** [comment]
Yeah, we only accept it for specific dependencies currently, we should fix that so that all dependencies accept at least vala. I think @eli-schwartz would really like to change the way rust dependencies work to be `dependency('foo', language : 'rust')` instead of `dependency('foo-rs')`
